### PR TITLE
feat(ocr-parser): Implementation for ocr-parser module 

### LINKED ISSUE #756: Contract parser client to extract contract clauses and details from a pdf/word document
This app contains APIs and modules to help finding and extracting contract clauses and details from a PDF/Word document, using a pretrained model, regex matching or text search.

### PR DESCRIPTION
**Pull request by vijay-webApps**
## Description
Implemented OCR-Parser to help finding and extracting contract clauses and details from a PDF document

Fixes #756 

## Type of change

Please delete options that are not relevant.

- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] Intermediate change (work in progress)

## How Has This Been Tested?

Please describe the tests that you ran to verify your changes. Provide instructions so we can reproduce. Please also list any relevant details for your test configuration

- [ ] Test A
- [ ] Test B

## Checklist:

- [x] Performed a self-review of my own code
- [x] npm test passes on your machine
- [ ] New tests added or existing tests modified to cover all changes
- [x] Code conforms with the style guide
- [ ] API Documentation in code was updated
- [ ] Any dependent changes have been merged and published in downstream modules

### CODE REVIEW

File: packages/ocr-parser/README.md [hunk 1]
```diff
@@ -0,0 +1,27 @@
+# OcrParser
```
Comment (akshatdubeysf) [inline packages/ocr-parser/README.md:1]: README?

File: packages/ocr-parser/projects/ocr-parser-lib/package.json [hunk 2]
```diff
@@ -0,0 +1,16 @@
+{
+  "name": "ocr-parser-lib",
```
Comment (akshatdubeysf) [inline packages/ocr-parser/projects/ocr-parser-lib/package.json:2]: name should follow sourceloop convention

Comment (akshatdubeysf) [inline packages/ocr-parser/projects/ocr-parser-lib/package.json:2]: name should follow sourceloop convention

File: packages/ocr-parser/projects/ocr-parser-lib/src/lib/ocr-parser/components/html-viewer/html-viewer.component.ts [hunk 3]
```diff
@@ -0,0 +1,66 @@
...
+  constructor(private readonly dataService: OcrDataService, @Inject(DOCUMENT) private document: Document) {
+
+  }
+
+  ngOnInit(): void {
+    this.subscription = this.dataService.$getSelectedClauseData.subscribe((data: FieldData) => {
+      if (data.value) {
+        this.fieldValue = data.value;
+        clearTimeout(timerId);
+        timerId = setTimeout(() => {
```
Comment (akshatdubeysf) [inline packages/ocr-parser/projects/ocr-parser-lib/src/lib/ocr-parser/components/html-viewer/html-viewer.component.ts:32]: why not use timeout from rxjs?

File: packages/ocr-parser/projects/ocr-parser-lib/src/lib/ocr-parser/components/html-viewer/html-viewer.component.ts [hunk 4]
```diff
@@ -0,0 +1,66 @@
...
+    }
+  }
+
+
+
+
+  scrollToFirstMarkedText() {
+    const markedElements = this.document.getElementsByTagName('mark'); // it will return the array of marked
+    if (markedElements.length) {
+      markedElements[0].scrollIntoView({ // scroll to first mark text
```
Comment (akshatdubeysf) [inline packages/ocr-parser/projects/ocr-parser-lib/src/lib/ocr-parser/components/html-viewer/html-viewer.component.ts:55]: Can this be written in a better way than to hardcode getting first index with mark tag?

File: packages/ocr-parser/projects/ocr-parser-lib/src/lib/ocr-parser/ocr-parser.component.ts [hunk 5]
```diff
@@ -0,0 +1,60 @@
+import { Component, EventEmitter, Input, OnDestroy, OnInit, Output } from '@angular/core';
+import { Subscription } from 'rxjs';
+import { DocumentConfig, FieldConfig, FieldData } from './models/ocr.model';
+import { OcrDataService } from './services/ocrData.service';
+
+@Component({
+  selector: 'sourceloop-ocrParser',
```
Comment (akshatdubeysf) [inline packages/ocr-parser/projects/ocr-parser-lib/src/lib/ocr-parser/ocr-parser.component.ts:7]: try to keep name hyphenated

File: packages/ocr-parser/projects/ocr-parser-lib/src/lib/ocr-parser/components/html-viewer/html-viewer.component.ts [hunk 6]
```diff
@@ -0,0 +1,92 @@
...
+ 
+	}
+
+  onUpdateClauseValue() {
+    this.dataService.setUpdatedClauseValue(this.fieldValue);
+  }
+
+  scrollToFirstMarkedText() {
+    const markedElements = this.document.getElementsByTagName('mark'); // it will return the array of marked
+    console.log(markedElements);
```
Comment (akshatdubeysf) [inline packages/ocr-parser/projects/ocr-parser-lib/src/lib/ocr-parser/components/html-viewer/html-viewer.component.ts:78]: No console.log

File: packages/ocr-parser/README.md (hunk 1 above)
Comment (vijay-webApps) [reply packages/ocr-parser/README.md]: Fixed

File: packages/ocr-parser/projects/ocr-parser-lib/package.json (hunk 2 above)
Comment (vijay-webApps) [reply packages/ocr-parser/projects/ocr-parser-lib/package.json]: Fixed

**Review by akshatdubeysf — CHANGES_REQUESTED** [review]

File: packages/ocr-parser/projects/ocr-parser-lib/package.json (hunk 2 above)
Comment (vijay-webApps) [reply packages/ocr-parser/projects/ocr-parser-lib/package.json]: Fixed

File: packages/ocr-parser/projects/ocr-parser-lib/src/lib/ocr-parser/components/html-viewer/html-viewer.component.ts (hunk 3 above)
Comment (vijay-webApps) [reply packages/ocr-parser/projects/ocr-parser-lib/src/lib/ocr-parser/components/html-viewer/html-viewer.component.ts]: Fixed

File: packages/ocr-parser/projects/ocr-parser-lib/src/lib/ocr-parser/components/html-viewer/html-viewer.component.ts (hunk 4 above)
Comment (vijay-webApps) [reply packages/ocr-parser/projects/ocr-parser-lib/src/lib/ocr-parser/components/html-viewer/html-viewer.component.ts]: Fixed

File: packages/ocr-parser/projects/ocr-parser-lib/src/lib/ocr-parser/ocr-parser.component.ts (hunk 5 above)
Comment (vijay-webApps) [reply packages/ocr-parser/projects/ocr-parser-lib/src/lib/ocr-parser/ocr-parser.component.ts]: Fixed

File: packages/ocr-parser/projects/ocr-parser-lib/src/lib/ocr-parser/components/html-viewer/html-viewer.component.ts (hunk 6 above)
Comment (vijay-webApps) [reply packages/ocr-parser/projects/ocr-parser-lib/src/lib/ocr-parser/components/html-viewer/html-viewer.component.ts]: Fixed

File: packages/ocr-parser/package.json [hunk 7]
```diff
@@ -0,0 +1,45 @@
...
+  "name": "@sourceloop/ocr-parser",
+  "version": "0.0.0",
+  "scripts": {
+    "ng": "ng",
+    "start": "ng serve",
+    "build": "ng build",
+    "watch": "ng build --watch --configuration development",
+    "test": "ng test"
+  },
+  "private": true,
```
[the remark this answers is not in the document]
Comment (vijay-webApps) [reply packages/ocr-parser/package.json]: Fixed

**Review by akshatdubeysf — APPROVED** [review]

**Pull request merged**